src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -69,6 +80,49 @@ def app_init(
     )
 
 
+@app.command("list-templates", hidden=True)
+def app_list_templates(**options) -> CommandResult:
+    """
+    Prints information regarding templates that can be used with snow app init.
+    """
+    with SecurePath.temporary_directory() as temp_path:
+        from git import Repo
+
+        repo = Repo.clone_from(
+            url=OFFICIAL_TEMPLATES_GITHUB_URL,
+            to_path=temp_path.path,
+            filter=["tree:0"],
+            depth=1,
+        )
```

Comment:
IMHO we should reuse the logic from the init command instead of reimplementing it here. You'll probably have to factor it out first though.

Reply:
Good callout, I extracted a function to `src/snowflake/cli/plugins/nativeapp/utils.py`, let me know if that works for you!